### PR TITLE
enhancement(vrl): Add `format_int` and `parse_int` functions

### PR DESCRIPTION
**Pull request by jszwedko**
For converting integers to/from strings.

We currently have `to_int` which is equivalent to `parse_int(foo, 10)`,
but the added `parse_int` is able to:

* Parse in a specified base
* Determine the base of the number by looking at its prefix (e.g. 0x ->
  16)

I added an analogous `format_int` for going the other way.

Closes: https://github.com/timberio/vector/issues/7228

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Comment by jszwedko** [comment]
@hhromic would you mind taking a peek and seeing if this solves your issue? Based on https://github.com/timberio/vector/issues/7228 it seems like it should.

File: docs/reference/remap/functions/format_int.cue [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+remap: functions: format_int: {
+	category: "Number"
+	description: #"""
+		Formats the integer `value` into a string representation using the given base/radix.
+		"""#
+
+	arguments: [
+		{
+			name:        "value"
+			description: "The number to format as a string."
```
Comment (spencergilbert) [inline docs/reference/remap/functions/format_int.cue:12]: As a string, but type = integer?

Comment (jszwedko) [reply docs/reference/remap/functions/format_int.cue:12]: True, `as` is ambiguous here 😄 I'll update it to just `The number to format`.

File: lib/vrl/stdlib/src/parse_int.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            },
+            Parameter {
+                keyword: "base",
+                kind: kind::INTEGER,
+                required: false,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
```
Comment (spencergilbert) [inline lib/vrl/stdlib/src/parse_int.rs:26]: Is it _legally_ required to always use 42 for example numbers 😛

**Review by spencergilbert — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**